src/hardware/gus.cpp

```diff
@@ -197,6 +195,88 @@ class Voice {
 static void GUS_TimerEvent(uint32_t t);
 static void GUS_DMA_Event(uint32_t val);
 
+// DRAM DMA Control Register (41h), section 2.6.1.1, page 12, of the UltraSound
+// Software Development Kit (SDK) Version 2.22
+//
+// This register is wrapped in a class because its state differs when written
+// versus read. When written, bit 6 indicates if the samples are 16-bit or
+// 8-bit. However when read, bit 6 indicates if a terminal count IRQ is pending.
+// There's no way for the application to read back the 16-bit sample indicator;
+// it's a one-way write into the GUS.
+//
+struct DmaControlRegister {
+public:
+	constexpr void Reset() noexcept
+	{
+		reg.data          = {};
+		are_samples_16bit = {};
+	}
+	constexpr void Set(const uint8_t data) noexcept
+	{
+		// Save our existing pending TC IRQ bit
```

Comment:
Maybe spell out "TC" in the comment.

Reply:
Good news - in the latest update, I was able to drop the class wrapper and use the bit-view register directly; so this `Set()` went away along with the need to save/load temp variables.